R/coverage.R

```diff
@@ -201,52 +201,49 @@ emdn_get_coverage <- function(
 
 # Convert coverage nil values to NA
 conv_nil_to_na <- function(cov_raster, summary, rangesubset) {
-  nil_values <- emdn_get_band_nil_values(summary)[rangesubset]
-  uniq_nil_val <- unique(nil_values)
-
-  # For efficiency, replace nil_values across entire coverage if all bands have
-  # the same nil_values. Return early.
-  if (length(uniq_nil_val) == 1L) {
-    if (is.numeric(uniq_nil_val)) {
-      terra::values(cov_raster)[
-        terra::values(cov_raster) >= uniq_nil_val
-      ] <- NA
-
-      cli_alert_success(
-        "nil values {.val {uniq_nil_val}} converted to {.field NA} on all bands."
+  purrr::reduce(
+    emdn_get_band_descriptions(summary),
+    \(cov_raster, x) {
+      conv_band_nil_value(
+        x,
+        cov_raster,
+        summary = summary,
+        rangesubset = rangesubset
       )
-    } else {
-      cli::cli_warn(
-        "!" = "Cannot convert non numeric nil value {.val {uniq_nil_val}} to NA"
-      )
-    }
-    return(cov_raster)
-  }
+    },
+    .init = cov_raster
+  )
+}
 
-  # If nil_values differ between bands, replace nil_values individually.
-  n_bands <- terra::nlyr(cov_raster)
-  nil_values <- rep(nil_values, times = n_bands / length(nil_values))
+conv_band_nil_value <- function(band, cov_raster, summary, rangesubset) {
+  nil_val <- emdn_get_band_nil_values(summary, band)[rangesubset]
```

Comment:
why do we have to subset with the range subset? Shouldn't the nil value be the same for the whole layer?

Reply:
Hmmm not sure why I set it up like that originally but your instinct makes sense to me. Might be worth investigating